### PR TITLE
storage: skip TestRestoreReplicas

### PR DESCRIPTION
**Pull request by tbg**
See #40351.

Release note: None

### CODE REVIEW

File: pkg/storage/client_raft_test.go [hunk 1]
```diff
@@ -333,6 +333,8 @@ func TestReplicateRange(t *testing.T) {
 func TestRestoreReplicas(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 
+	defer t.Skip("https://github.com/cockroachdb/cockroach/issues/40351")
```
Comment (nvanbenschoten) [inline pkg/storage/client_raft_test.go]: Why are you deferring this?

Comment (tbg) [reply pkg/storage/client_raft_test.go]: Because I was braindead, thanks for noticing

**Comment by tbg** [comment]
Looks like you didn't approve, so I'm going to hold off on the merge.

**Review by knz — APPROVED** [review]
thanks

**Comment by tbg** [comment]
Thanks @knz!

bors r=knz

**Pull request merged**